app/models/create_cap_proposal.rb

```diff
@@ -0,0 +1,92 @@
+class CreateCapProposal
+  class Error < StandardError; end
+
+  OFFICE = 'DC'
+  PURCHASE_TYPE = 'Software'
+  QUANTITY = 1
+  RECURRING = false
+  URGENCY = 20
+
+  def initialize(auction)
+    @auction = auction
+  end
+
+  def configure_c2_client
+    @host = ENV.fetch('C2_HOST', 'https://cap.18f.gov')
+    @c2_client = C2::Client.new(
+      oauth_key: ENV.fetch('C2_OAUTH_KEY'),
+      oauth_secret: ENV.fetch('C2_OAUTH_SECRET'),
+      host: @host,
+      debug: ENV.fetch('C2_DEBUG', false)
+    )
+
+    @c2_client
+  end
+
+  def c2_client
+    @c2_client ||= configure_c2_client
+  end
+
+  def perform
+    begin
+      resp = c2_client.post('proposals', c2_proposal_hash)
+      proposal = resp.body
+      return construct_proposal_url(proposal.id)
+    rescue error => Faraday::ClientError
+      fail CreateCapProposal::Error, error
+    end
+  end
+
+  def c2_proposal_hash
+    {
+      office: CreateCapProposal::OFFICE,
+      purchase_type: CreateCapProposal::PURCHASE_TYPE,
+      product_name_and_description: product_name_and_description,
+      justification: justification,
+      link_to_product: link_to_product,
+      cost_per_unit: cost_per_unit,
+      quantity: CreateCapProposal::QUANTITY,
+      recurring: CreateCapProposal::RECURRING,
+      date_requested: date_requested,
+      urgency: CreateCapProposal::URGENCY,
+      additional_info: additional_info
+    }
+  end
+
+  def product_name_and_description
+    %{Micropurchase for '#{@auction.title}'.
+
+      Link: #{@auction.url}
+
+      Summary:
+
+      #{@auction.summary}}
+  end
+
+  def link_to_product
+    @auction.delivery_url
+  end
+
+  def justification
+    "Billable to the '#{@auction.billable_to}' Tock line item."
+  end
+
+  def cost_per_unit
+    @auction.winning_bid.amount
+  end
+
+  def date_requested
+    Date.today.iso8601
```

Comment:
consider Date.current (instead of Date.today) -- current is in the timezone of the application ... research this.